### PR TITLE
Fix TablePrivacyManager duplicated code

### LINKED ISSUE #2501: Fix TablePrivacyManager duplicated code
Part of an ongoing cache-related cleaning spree. Remove some duplicated code from table privacy manager (and move one of the vizjson key methods of Vis member near the other).

### PR DESCRIPTION
**Pull request by Kartones**
Fixes #2501

### CODE REVIEW

**Comment by Kartones** [comment]
@rafatower  CR and so that you know

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/1926/))

File: app/models/table.rb [hunk 1]
```diff
@@ -702,9 +702,9 @@ def remove_table_from_user_database
 
   # This method removes all the vanish cached objects for the table,
   # tiles included. Use with care O:-)
-  def invalidate_varnish_cache
+  def invalidate_varnish_cache(propagate_to_visualizations=true)
```
Comment (rafatower) [inline app/models/table.rb]: I'd rename this to `invalidate_cache`. Note that `invalidate_cache_for(visualizations)` also invalidates redis. From outside the `Visualization::member` I guess we don't want to leak whether it is varnish or redis or both.

**Comment by rafatower** [comment]
Also it lacks a test, I can help you with it

**Comment by Kartones** [comment]
Tested at pre-production environment

**Pull request merged**